.github/workflows/build_preview_website.yaml

```diff
@@ -22,6 +22,8 @@ jobs:
     steps:
       - name: Checkout Repo
         uses: actions/checkout@v3
+        with:
+          ref: ${{ github.event.pull_request.head.sha }}
```

Comment:
What is this for?

Reply:
I was researching the reason for the workflows not building the latest website versions like here: https://github.com/fluttercommunity/plus_plugins/pull/1389#issuecomment-1345403167

Same thing happened in this PR as well and, I guess, in any other PR we had so far for this workflow (potentially, for all other workflows as well). To see that not the latest version is built see logs for the `build website preview` workflow:
<img width="814" alt="Screenshot 2022-12-12 at 12 08 13" src="https://user-images.githubusercontent.com/13467769/207018869-8cea22ec-21a4-49e1-bd7f-fe6b23c2b14b.png">

As you might see the website version is still `0.2.1` even though I bumped it in this PR as well as older Docusaurus dependencies are used.

While looking through logs I found out that the issue is in checkout action, which checks out latest commit from the `main` branch, not the latest commit from the PR. 

As it turned out it happens for workflows that get restarted via `Re-run jobs`, for example: https://github.com/actions/checkout/issues/1036

Thus, I am experimenting with workflow updates to see what will fix the issue for website preview builds for us. Additionally, as we use checkout action everywhere all other workflows are affect by the same issue as well. So as soon as I find out the solution I will create another PR to fix the issue.

Reply:
Cool, thanks for the explanation ❤️

Reply:
As it turned out the issue with `checkout` happens only when workflow is triggered by event like `pull_request_target` which we use for this preview deployment workflow only. Thus, my theory about other workflows being affected as well isn't correct.